src/expr/algorithms/flatten.h

```diff
@@ -0,0 +1,121 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Gereon Kremer
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Utilities for flattening nodes.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__EXPR__ALGORITHMS__FLATTEN_H
+#define CVC5__EXPR__ALGORITHMS__FLATTEN_H
+
+#include <algorithm>
+
+#include "expr/node.h"
+
+namespace cvc5::expr {
```

Comment:
I think that it would be a bit more consistent to move this to the `cvc5::expr::algorithms` namespace. I also have a slight preference for `algorithm` over `algorithms`. What do you think?

```suggestion
namespace cvc5::expr::algorithms {
```